tt_metal/hw/inc/ethernet/dataflow_api.h

```diff
@@ -65,6 +61,63 @@ void check_and_context_switch() {
     // proceed
 }
 
+FORCE_INLINE
+void disable_erisc_app() { flag_disable[0] = 0; }
+
+FORCE_INLINE
+void send_fd_packets() {
+    eth_send_packet(
+        0,
+        (eth_l1_mem::address_map::ERISC_APP_RESERVED_BASE) >> 4,
+        ((eth_l1_mem::address_map::ERISC_APP_RESERVED_BASE)) >> 4,
+        (eth_l1_mem::address_map::ERISC_APP_RESERVED_SIZE) >> 4);
+    routing_info->fd_buffer_msgs_sent = 1;
+    eth_send_packet(
+        0,
+        ((uint32_t)(&(routing_info->fd_buffer_msgs_sent))) >> 4,
+        ((uint32_t)(&(routing_info->fd_buffer_msgs_sent))) >> 4,
+        1);
+    // There should always be a valid cmd here, since eth_db_acquire completed
+    while (routing_info->fd_buffer_msgs_sent != 0) {
+        // TODO: add timer to restrict this
+        risc_context_switch();
+    }
+}
+
+FORCE_INLINE
+void receive_fd_packets() {
+    // There may not be a valid cmd here, since DST router is always polling
+    // This should only happen on cluster close
+    while (routing_info->fd_buffer_msgs_sent != 1 && routing_info->routing_enabled) {
+        // TODO: add timer to restrict this
+        risc_context_switch();
+    }
+    routing_info->fd_buffer_msgs_sent = 0;
+    eth_send_packet(
+        0,
+        ((uint32_t)(&(routing_info->fd_buffer_msgs_sent))) >> 4,
+        ((uint32_t)(&(routing_info->fd_buffer_msgs_sent))) >> 4,
+        1);
+}
+
+}  // namespace internal_
+
+void run_routing() {
+    // TODO: maybe split into two FWs? or this may be better to sometimes allow each eth core to do both send and
+    // receive of fd packets
+    if (routing_info->routing_mode == EthRouterMode::FD_SRC) {
+        // TODO: port changes from erisc to here
```

Comment:
is this TODO part of this PR?

Reply:
No, it's more refactor work so things look nicer. I may clean it up in the next phase of integrating with your remote processor changes. I just avoided refactoring in case this dramatically changes